lib/spack/spack/test/conftest.py

```diff
@@ -725,11 +725,31 @@ def mock_archive(request, tmpdir_factory):
 
 @pytest.fixture(scope='session')
 def mock_git_repository(tmpdir_factory):
-    """Creates a very simple git repository with two branches and
-    two commits.
+    """Creates a very simple git repository with two branches,
+    two commits and five submodules.
     """
     git = spack.util.executable.which('git', required=True)
 
+    suburls = []
+    for submodule_count in range(5):
```

Comment:
A few minor requests:

* Generate fewer submodules: as many as are necessary to test what you need. I think two would be sufficient.
* Generate different contents in each submodule (can just be different file names)
* Mention in the docstring that each submodule has one commit
* In the docstring "very simple" -> "simple" (there are simpler git repos)

Reply:
Done